### PR TITLE
Handle the case where Konductor responds without an identity

### PR DESCRIPTION
**Pull request by jonsnyder**
## Description

This fixes one part of the Jira Ticket. The changes make it so that if the first request without an ecid failed to get one, the next request will be sent and request the ecid. I will add addition changes and tests to handle errors differently as outlined in the Jira ticket to a separate PR.

## Related Issue

https://jira.corp.adobe.com/browse/CORE-47161

## Motivation and Context

Previously if Konductor came back without an identity, no other requests could be sent through Alloy. You would have to do a page refresh to start making requests again. This was found when we were testing the validation logic for consent. If an invalid consent object was provided, Konductor returned an error without identity information.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have made any necessary test changes and all tests pass.
- [ ] I have run the Sandbox successfully.

### CODE REVIEW

File: src/components/Identity/injectEnsureRequestHasIdentity.js [hunk 1]
```diff
@@ -18,38 +18,72 @@ export default ({
...
+   * will have the newly minted ECID.
+   *
+   * Konductor should make every effort to return an identity, but in
+   * certain scenarios it may not. For example, in cases where the
+   * request does not match what Konductor is expecting (ie 400s).
+   * In cases where Konductor does not set an identity, there should be
+   * no events recorded so we don't need to worry about multiple ECIDs
+   * being minted for each user.
+   *
+   * If we only allowed one request through without an identity, a single
```
Comment (liljenback) [inline src/components/Identity/injectEnsureRequestHasIdentity.js:43]: Maybe reword this a bit to say something like: the reason we allow for multiple sequential requests to be sent without an identity is to prevent a single malformed ...

File: src/components/Identity/injectEnsureRequestHasIdentity.js [hunk 2]
```diff
@@ -18,38 +18,72 @@ export default ({
...
+      return (
+        previousObtainedIdentityPromise
+          .then(() => {
+            logger.log("Resuming previously delayed request.");
+          })
+          // If Konductor did not set the identity cookie on the previous
+          // request, then awaitIdentityCookie will reject its promise.
+          // Catch the rejection here and allow this request to go out.
+          .catch(() => {
+            return allowRequestToGoWithoutIdentity(payload);
```
Comment (liljenback) [inline src/components/Identity/injectEnsureRequestHasIdentity.js:77]: What about the requests that doesn't result in errors?

Comment (jonsnyder) [reply src/components/Identity/injectEnsureRequestHasIdentity.js:77]: Requests that don't result in errors would trigger the then clause, and so the promise would resolve after logging. Once the promise resolves, the request would go out. Is there a particular use case you are thinking about?

Comment (liljenback) [reply src/components/Identity/injectEnsureRequestHasIdentity.js:77]: But does it get sent (this would be request 2,3,etc) if the first req doesn't end in error? What am I missing?

**Comment by jfkhoury** [comment]
> ## Description
> This fixes one part of the Jira Ticket. The changes make it so that if the first request without an ecid failed to get one, the next request will be sent and request the ecid. I will add addition changes and tests to handle errors differently as outlined in the Jira ticket to a separate PR.
> 
> ## Related Issue
> https://jira.corp.adobe.com/browse/CORE-47161
> 
> ## Motivation and Context
> Previously if Konductor came back without an identity, no other requests could be sent through Alloy. You would have to do a page refresh to start making requests again. This was found when we were testing the validation logic for consent. If an invalid consent object was provided, Konductor returned an error without identity information.
> 
> ## Screenshots (if appropriate):
> ## Types of changes
> * [x]  Bug fix (non-breaking change which fixes an issue)
> * [ ]  New feature (non-breaking change which adds functionality)
> * [ ]  Improvement (non-breaking change which does not add functionality)
> * [ ]  Breaking change (fix or feature that would cause existing functionality to change)
> 
> ## Checklist:
> * [x]  I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
> * [x]  I have made any necessary test changes and all tests pass.
> * [ ]  I have run the Sandbox successfully.

File: src/components/Identity/injectEnsureRequestHasIdentity.js (hunk 2 above)
Comment (jonsnyder) [reply src/components/Identity/injectEnsureRequestHasIdentity.js:77]: Yes it gets sent. The Identity component calls ensureRequestHasIdentity in the "onBeforeRequest" lifecycle method and returns the promise.  This lifecycle method waits for all the promises returned from "onBeforeRequest" to resolve before sending the request. So when the first request comes back with the identity, the promise resolves and the others are able to go out.

Comment (liljenback) [reply src/components/Identity/injectEnsureRequestHasIdentity.js:77]: 👍

**Comment by liljenback** [comment]
Maybe rename the ensure request has identity to ensure request has unique identity (or similar)
